Onboarding/engineering-best-practices.md

```diff
@@ -0,0 +1,36 @@
+# Engineering Best Practices
+
+Here is a brief list of Veteran Services Platform (VSP) recommended engineering best practices when working to add new features to the VSP.  
+
+### Tickets/Issues/User stories
+
+- refine tickets until they are atomic (possible guideline: nothing larger than an 8)
+- use epics to tether together common bodies of atomic work
+- include a background section providing context, resources, links, images, and/or POCs
+- identify clear, concise, achievable acceptance criteria
+- when applicable, identify risks that could impact delivery, resources, quality, etc.
+- use tickets as a forum to discuss and capture implementation ideas, questions, patterns, access, etc.
+  - in particular when implementation path is not clear, favor discussion prior to submitting a PR 
+- when applicable, use tickets to collaborate and gain consensus on API contracts
+- subscribe relevant VSP team members for visibility and agile support
+- maintain accurate ticket status details (i.e. pipelines, assignees, blockers, etc.)
+
+### Pull requests
+
+- do not go longer than a week without pushing a WIP PR
+- should there be scope creep, favor multiple, smaller, focused PR's over large PR's 
+- include a background section providing context and a summary of why the PR is necessary, and what you've done in the PR
+  - if applicable, include resources, links, images, and/or POCs
+- include a _Definition of Done_ section that parallels what was completed in the PR to the originating issue's _Acceptance Criteria_
+- when applicable, include screenshots (i.e. new API endpoint docs, new UI, etc.)
+- feel free to request reviews for a WIP PR, and then use GitHub's [re-request review feature](https://github.blog/changelog/2019-02-21-re-request-review-on-a-pull-request/) to get final PR approvals
+- identify WIP PR's by adding _WIP_ to the title, or using GitHub's [draft pull request feature](https://github.blog/2019-02-14-introducing-draft-pull-requests/)
+
+### Backend development
+
+- discuss your implementation ideas with VSP backend team members early and often; there are many common patterns and classes that are in place in Vets-API that should be leveraged, that we can point you to
+- use [VCR cassettes](https://github.com/vcr/vcr) for recording external API responses
+- if integrating a new external API, at the onset, request VPS DevOps team to make connections between respective Vets-API & external API environments (i.e. staging to staging, etc.)
+
+
+### Frontend development
```

Comment:
@jbalboni @rianfowler @accbjt - 👋 

For the **Frontend development** section, can you please provide any relevant "talking points" regarding:

- What can/should be said here around "Feature-flag-based development"
- What else should be included here?

Thanks!

Reply:
Here's what I have. I tried to match it to what was said about the BE stuff since there's a lot of material we could cover. Happy to discuss this further. 

- Familiarize yourself with the [Client application documentation](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/getting-started) site. It summarizes several common tasks, common components, and the high level structure of `vets-website` 
- The master branch of `vets-website` is deployed [every weekday](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/platform/site-structure/environments/). We encourage frequent merges of small pull requests and using [environment flags](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/platform/tools/feature-flags/) to keep partially complete or in progress features from appearing in production. 
  - If modifying an existing app, we encourage writing code that's [easy to delete](https://programmingisterrible.com/post/139222674273/write-code-that-is-easy-to-delete-not-easy-to). We encourage coarse feature gating logic (i.e. "please repeat yourself") to avoid needing "shotgun surgery" later when the feature goes live. 
  - We rely heavily on automated testing in our continuous integration flow to detect regressions. It's important to follow our unit testing and end to end testing patterns to maintain a highly collaborative environment. 
- VSP front end engineers are available to review or discuss changes.

Reply:
This is great...thanks!